Revit_Core_Engine/Convert/Structure/FromRevit/Profile.cs

```diff
@@ -171,18 +174,19 @@ private static FabricatedISectionProfile FabricatedISectionProfileFromRevit(this
             }
             else
             {
-                height = familySymbol.LookupParameterDouble(heightNames);
-                topFlangeWidth = familySymbol.LookupParameterDouble(topFlangeWidthNames);
-                botFlangeWidth = familySymbol.LookupParameterDouble(botFlangeWidthNames);
-                webThickness = familySymbol.LookupParameterDouble(webThicknessNames);
-                topFlangeThickness = familySymbol.LookupParameterDouble(topFlangeThicknessNames);
-                botFlangeThickness = familySymbol.LookupParameterDouble(botFlangeThicknessNames);
-                weldSize = familySymbol.LookupParameterDouble(weldSizeNames1);
+                BH.Engine.Reflection.Compute.RecordWarning($"Dimensions of section profile {familySymbol.Name} (Revit ElementId {familySymbol.Id.IntegerValue}) were extracted using name matching and may be incorrect in some cases.");
+                height = familySymbol.LookupParameterDouble(heightNames, dimensionGroups);
+                topFlangeWidth = familySymbol.LookupParameterDouble(topFlangeWidthNames, dimensionGroups);
+                botFlangeWidth = familySymbol.LookupParameterDouble(botFlangeWidthNames, dimensionGroups);
+                webThickness = familySymbol.LookupParameterDouble(webThicknessNames, dimensionGroups);
+                topFlangeThickness = familySymbol.LookupParameterDouble(topFlangeThicknessNames, dimensionGroups);
+                botFlangeThickness = familySymbol.LookupParameterDouble(botFlangeThicknessNames, dimensionGroups);
+                weldSize = familySymbol.LookupParameterDouble(weldSizeNames1, dimensionGroups);
             }
 
             if (double.IsNaN(weldSize))
             {
-                weldSize = familySymbol.LookupParameterDouble(weldSizeNames2);
+                weldSize = familySymbol.LookupParameterDouble(weldSizeNames2, dimensionGroups);
                 if (!double.IsNaN(weldSize) && !double.IsNaN(webThickness))
                     weldSize = (weldSize - webThickness) / (Math.Sqrt(2));
```

Comment:
Found in testing, think this should be:

```suggestion
                    weldSize = (weldSize - webThickness / 2) / (Math.Sqrt(2));
```